content/blog/2021/11/deep-dive-into-policy-logging.md

```diff
@@ -0,0 +1,218 @@
+---
+title: Deep Dive into policy logging
+authors:
+    - Rafael Fernández López
+date: 2021-11-15
+---
+
+Policies are regular programs. As such they have the need to log often times. In general, we are
+used to make our programs log into standard output (stdout) and standard error (stderr) outputs.
+
+However, policies run in a confined WebAssembly environment. For this mechanism to work as usual
+Kubewarden would need to set up the runtime environment in a way that stdout and stderr file
+descriptors are set up so that the policy can write to them, and upon completion, Kubewarden can
+check them -- or stream them as they pop up.
+
+Although the previous approach was our very first initial implementation in Kubewarden, it's no
+longer implemented that way.
+
+Given we are providing SDK's for supported languages we can do a bit better. As we have described in
+previous posts and in our documentation, currently Kubewarden uses waPC for allowing
+intercommunication between the guest (the policy) and the host (Kubewarden -- the `policy-server` or
+`kwctl` if we are running policies manually).
+
+Kubewarden already sets up a contract between the policies (guests) and the host (Kubewarden) for
+performing [policy settings validation](https://docs.kubewarden.io/writing-policies/spec/02-settings.html), [policy validation](https://docs.kubewarden.io/writing-policies/spec/03-validating-policies.html), [policy mutation](https://docs.kubewarden.io/writing-policies/spec/04-mutating-policies.html) and logging.
+
+The waPC interface used for logging is a contract, because once you have built a policy, it should
+be possible to run it in future Kubewarden versions. In this sense, we have preferred to keep this
+contract behind the SDK of your preferred language, so you don't have to deal with the details of
+how logging is implemented in Kubewarden. You just have to use your preferred logging library for
+the language of your choice.
+
+Let's look into how to take advantage of logging with Kubewarden in specific languages!
+
+## Policy Authors
```

Comment:
Nit:

```suggestion
## For policy Authors
```
(And the same in "For cluster administrators")

Just so it feels warmer.